test/WormholeReceiver.t.sol

```diff
@@ -17,22 +17,71 @@ contract WormholeReceiverTestHarness is WormholeReceiver {
     bytes32 deliveryHash
   ) public virtual override {}
 
-  function modifierTest() public onlyRelayer {}
+  function onlyRelayerModifierFunc() public onlyRelayer {}
+
+  function isRegisteredSenderModifierFunc(uint16 sourceChain, bytes32 senderAddress)
+    public
+    isRegisteredSender(sourceChain, senderAddress)
+  {}
+}
+
+contract WormholeReceiverTest is Test, Constants {
+  WormholeReceiverTestHarness receiver;
+
+  function setUp() public {
+    receiver = new WormholeReceiverTestHarness(L1_CHAIN.wormholeRelayer);
+  }
 }
 
-contract OnlyRelayerTest is Test, Constants {
-  function testFuzz_SucceedsIfCalledByWormholeRelayer(address relayer) public {
+contract OnlyRelayer is Test, Constants {
+  function testFuzz_SucceedIfCalledByWormholeRelayer(address relayer) public {
     WormholeReceiverTestHarness receiver = new WormholeReceiverTestHarness(relayer);
 
     vm.prank(relayer);
-    receiver.modifierTest();
+    receiver.onlyRelayerModifierFunc();
   }
 
   function testFuzz_RevertIf_NotCalledByWormholeRelayer(address relayer) public {
     vm.assume(relayer != address(this));
     WormholeReceiverTestHarness receiver = new WormholeReceiverTestHarness(relayer);
 
     vm.expectRevert(WormholeReceiver.OnlyRelayerAllowed.selector);
-    receiver.modifierTest();
+    receiver.onlyRelayerModifierFunc();
+  }
+}
+
+contract SetRegisteredSender is WormholeReceiverTest {
+  function testFuzz_SuccessfullySetRegisteredSender(uint16 sourceChain, address sender) public {
+    bytes32 senderBytes = bytes32(uint256(uint160(address(sender))));
+    receiver.setRegisteredSender(sourceChain, senderBytes);
+
+    assertEq(
+      receiver.registeredSenders(sourceChain, senderBytes),
+      true,
+      "Registered sender on source chain is not correct"
+    );
+  }
+}
+
+contract IsRegisteredSender is WormholeReceiverTest {
+  function testFuzz_SuccessfullyCallWithRegisteredSender(uint16 sourceChain, address sender) public {
+    vm.assume(sender != address(0));
+    bytes32 senderBytes = bytes32(uint256(uint160(address(sender))));
+    receiver.setRegisteredSender(sourceChain, senderBytes);
+    receiver.isRegisteredSenderModifierFunc(sourceChain, senderBytes);
+  }
+
+  function testFuzz_RevertIf_NotCalledByRegisteredSender(
```

Comment:
I have been seeing intermittent errors here. I haven't been able to reproduce consistently locally.

Reply:
>  I haven't been able to reproduce consistently locally.

Wdym here? If we have a failure, foundry reports the inputs that triggered the failure, and if you create a concrete test with those value it should now always fail. Are you saying that's flaky, e.g. maybe dependent on some fork state?

Reply:
Does the failure also occur locally? Github redacts repo secrets in logs, so each `***` means that value matched some repo secret we have:
- https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions
- https://github.com/actions/runner/issues/507#issuecomment-1302424205

Some of them do seem weird, like `**m*** failed`. You can also use [act](https://github.com/nektos/act) to run GH actions locally and maybe that will help

Reply:
Cool, will give it a try. I have seen some failures locally but they are very intermittent and sometimes on other branchs. Looks like it just failed now so going to try to reproduce locally.

Reply:
Ed's pr should handle this